be/src/io/fs/s3_file_system.h

```diff
@@ -121,7 +121,7 @@ class S3FileSystem final : public RemoteFileSystem {
             abs_path = path;
         } else {
             // path with no schema
-            abs_path = _root_path / path;
+            abs_path = _prefix / path;
```

Comment:
_prefix is normalized in S3FileSystem's constructor, removed the first and last '/'